### PR TITLE
Adding a shell layer to find python version on the box

### PR DESCRIPTION
**Pull request by vrdmr**
Added a new Shell script to find the correct python version and invoke the extension script.

FYI: @narrieta, @boumenot, @roiyz-msft.

### CODE REVIEW

File: VMAccess/vmaccess.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
```
Comment (narrieta) [inline VMAccess/vmaccess.sh]: If this approach is going to be used by other extensions, consider making this a more generic script that takes as argument the python script (with its arguments) that needs to be executed.

File: VMAccess/vmaccess.sh [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    -d|--disable    disable the extension
+    -e|--enable     enable the extension
+    -p|--update     update the extension
+"
+
+function find_python(){
+    local python_version=$1
+
+    # Check if there is python defined.
+    which python > /dev/null 2>&1
```
Comment (narrieta) [inline VMAccess/vmaccess.sh]: There are some posts (e.g. https://stackoverflow.com/questions/592620/how-to-check-if-a-program-exists-from-a-bash-script) discouraging the use of which to check for the existence of a command; apparently some implementations don't set an exit code. Consider using command instead:
```

if command -v python; then
...
fi
```

File: VMAccess/vmaccess.sh [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    return_code=`echo $?`
+
+    if [[ ${return_code} == 0 ]]
+    then
+        eval ${python_version}="python"
```
Comment (narrieta) [inline VMAccess/vmaccess.sh]: The name of the variable 'python_version' is a little misleading... 'python_command' maybe?

File: VMAccess/vmaccess.sh [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            exit 1
+            ;;
+    esac
+done
+
+shift $((OPTIND-1))
+
+find_python PYTHON
+
+if [ -z "$PYTHON" ]
```
Comment (narrieta) [inline VMAccess/vmaccess.sh]: find_python is also doing an 'exit 51'; we should probably remove that one

File: VMAccess/vmaccess.sh [hunk 5]
```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+
+COMMAND="./vmaccess.py"
+PYTHON=""
+
+USAGE="$(basename "$0") [-h] [-i|--install] [-u|--uninstall] [-d|--disable] [-e|--enable] [-p|--update] -- program to invoke the extension script
```
Comment (narrieta) [inline VMAccess/vmaccess.sh]: consider moving the description of the script "program to invoke the extension script" to another line to separate it from the syntax

File: VMAccess/manifest.xml [hunk 6]
```diff
@@ -2,7 +2,7 @@
 <ExtensionImage xmlns="http://schemas.microsoft.com/windowsazure">
   <ProviderNameSpace>Microsoft.OSTCExtensions</ProviderNameSpace>
   <Type>VMAccessForLinux</Type>
-  <Version>1.5.0</Version>
+  <Version>1.6.0</Version>
```
[the remark this answers is not in the document]
Comment (vrdmr) [reply VMAccess/manifest.xml]: Done.

File: VMAccess/vmaccess.sh (hunk 1 above)
Comment (vrdmr) [reply VMAccess/vmaccess.sh]: Good idea. I added a new `-c` flag where one can give the python script. Please take a look at the update `$USAGE` for its usage.

**Comment by narrieta** [comment]
thanks for the updates, lgtm

File: VMAccess/HandlerManifest.json [hunk 7]
```diff
@@ -2,11 +2,11 @@
   {
     "version": 1.0,
     "handlerManifest": {
-      "disableCommand": "./vmaccess.py -disable",
-      "enableCommand": "./vmaccess.py -enable",
-      "installCommand": "./vmaccess.py -install",
-      "uninstallCommand": "./vmaccess.py -uninstall",
-      "updateCommand": "./vmaccess.py -update",
+      "disableCommand": "./extension_shim.sh -c ./vmaccess.py -d",
```
[the remark this answers is not in the document]
Comment (vrdmr) [reply VMAccess/HandlerManifest.json]: okay. I'll change this. Thanks.

**Pull request merged**